includes/class-wc-payments.php

```diff
@@ -668,7 +668,10 @@ public static function register_gateway( $gateways ) {
 			} else {
 				self::$registered_card_gateway = self::$card_gateway;
 			}
-			$gateways[]       = self::$registered_card_gateway;
+			$gateways[] = self::$registered_card_gateway;
+			if ( self::$registered_card_gateway->is_subscriptions_enabled() && self::$registered_card_gateway->is_changing_payment_method_for_subscription() ) {
+				return $gateways;
+			}
```

Comment:
```suggestion
			$gateways[] = self::$registered_card_gateway;
```
Correct me if I'm mistaken, but is the idea here to only return the card gateway when processing some kind of subscription payment? I think the safer play would be to add this logic down [here](https://github.com/Automattic/woocommerce-payments/blob/feature/deferred-upe-my-account-page/includes/class-wc-payments.php#L692). This seems like a more appropriate place for it and means that we won't need to revisit this block of code when we enable SEPA as a reusable payment method that can be used with subscriptions. What say?

Reply:
Thanks for the catch - this totally makes sense. I've updated the code in cfa0f6c06b5eb684d7d0adb0ac42366c5d64ad0d.